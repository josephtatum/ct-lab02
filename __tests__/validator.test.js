const { 
  nameValidator,
  ageValidator,
  weightValidator
} = require('../lib/validator.js');


describe('validator module', () => {
  it('shoud tell us if an object is validated', () => {
    expect(nameValidator.validate({ name: 'spot' })).toEqual('spot');
    expect(ageValidator.validate({ age: '5' })).toEqual(5);
    expect(weightValidator.validate({ weight: '20 lbs' })).toEqual('20 lbs');
  });

});
