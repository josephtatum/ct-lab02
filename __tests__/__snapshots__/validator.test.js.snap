// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validator module should throw an error if the field is required and is the wrong type 1`] = `"Cannot cast >>() => {}<< to String"`;

exports[`validator module should throw an error if the field is required and is the wrong type 2`] = `"Cannot cast >>() => {}<< to Number"`;

exports[`validator module should throw an error if the field isn't required and is the wrong type 1`] = `"Cannot cast >>() => {}<< to String"`;

exports[`validator module should throw an error if the object doesn't inlude a required key/value pair 1`] = `"error"`;
